crates/red_knot/src/types/infer.rs

```diff
@@ -119,12 +120,30 @@ where
     }
 }
 
+fn get_class_member<Db>(db: &Db, class_id: ClassTypeId, name: &Name) -> QueryResult<Option<Type>>
```

Comment:
Two questions:

1. Should this recurse up into superclasses to search for the member in those class bodies, if it can't find them in the class itself? I'm anticipating there are three possible answers to this: "yes", "no (we'll do the proper recursive searching through the mro somewhere else)", and "yes, but not now". Whichever answer you choose, I think it would be good to add a comment noting the behaviour we choose and explaining why we've chosen it ;)
2. How are we distinguishing between class variables and instance variables here? I can think of two possible solutions (though there may be more):
   1. Rather than returning a `QueryResult<Option<Type>>`, the function could return `QueryResult<Option<ClassMember>>`, where `ClassMember` is an enum:
      ```rs
      enum ClassMember {
          /// e.g. `x: int` at the class scope,
          /// or `self.${var}` assignments within methods
          InstanceMember(Type),
          /// e.g. `x: ClassVar[int]` at the class scope,
          /// method definitions, or `x = 42` in the class scope
          ClassMember(Type),
      }
      ```
   2. We could have two different functions, one for retrieving class variables and the other for retrieving instance variables.
   
   Again, it's okay if this question is deferred for now to keep things simple, but some kind of TODO comment might be in order.

Reply:
Great points!

> Should this recurse up into superclasses

I think we will need both a method that does, and a method that doesn't (which the method that does will need to use). So I think it's fine to first add the method that doesn't, though when I actually add the override-check lint rule, it may make sense to add the recursive lookup method then.

I'll add a comment and/or rename this method for better clarity.

> How are we distinguishing between class variables and instance variables here?

Currently we aren't, and we will need to at some point. It's good to at least consider now how we will want to.

I think one tricky thing here is that there are really three variants: instance member, class member, and instance-or-class member. (`x = 2` in class body without a `ClassVar` annotation is the latter, in general, though we may want to do additional analysis of whether any method includes `self.x = ...` to possibly narrow it down to "class member" if not.)

Because of that, I think I'm inclined to do this as two methods, where for some names, both methods would return the result? But I'm not sure, and I think it can be deferred. I'll add a TODO.